stardict.cc

```diff
@@ -466,22 +466,22 @@ string StardictDictionary::handleResource( char type, char const * resource, siz
     {
       QString articleText = QString( "<div class=\"sdct_h\">" ) + QString::fromUtf8( resource, size ) + "</div>";
 
+      const QString srcRegexpStr(
+          "(<\\s*(?:img|script)\\s[^>]*src\\s*=)(?!\\s*[\"']?(?:data|https?|ftp):)(\\s*[\"']?)" );
+      const QString linkRegexpStr(
+          "(<\\s*link\\s[^>]*href\\s*=)(?!\\s*[\"']?(?:data|https?|ftp):)(\\s*[\"']?)" );
```

Comment:
A possible regression because of the `InvertedGreedinessOption`/`setMinimal` removal: if there is an `src` attribute followed by something like `data-src` attribute, the `data-src`'s value will be replaced. Is a whitespace necessary before the attribute name in HTML? If so, maybe enforce it somehow? Something like `link\\s(?:[^>]*\\s)?href`

Reply:
Reasonable. I will take a look.

Also both the original and the renewed regex match `<img data-src='value'/>`

Reply:
Yes, they both match `data-src`. But I was looking for something that can possibly break by merging this PR and found the possible ordering that causes a regression :)

Reply:
Fixed with `(?:\s[^>]*\s|\s)` group, so both an element name and an attribute must be surrounded with whitespaces. First alternative matches `name··attribute` and also `name·someattr='...'·attribute` cases, and the second matches `name·attribute`.

Reply:
I think your group is equivalent to but more verbose than `\s(?:[^>]*\s)?` (repeats the same `\s` twice)